src/frontend/hooks/useAwaited.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useState } from 'react'
+
+export function useAwaited<T>(getter: () => Promise<T>): T | null
+export function useAwaited<T>(getter: () => Promise<T>, defaultValue: T): T
+export function useAwaited<T>(
+  getter: () => Promise<T>,
+  defaultValue: T | null = null
+): T | null {
+  const [value, setValue] = useState<T | null>(defaultValue)
+
+  // This is `setValue` as long as the component requesting the value is mounted
+  let setValueIfMounted = setValue
```

Comment:
I think this `let` can be moved inside the `useEffect` so it's not set for every render?

like:

```ts
 useEffect(() => {
    let setValueIfMounted = setValue
    getter().then(setValueIfMounted)
   ...
}
```